robot-server/robot_server/runs/run_store.py

```diff
@@ -442,28 +492,48 @@ def _clear_caches(self) -> None:
 def _convert_row_to_run(
     row: sqlalchemy.engine.Row,
     action_rows: List[sqlalchemy.engine.Row],
-) -> RunResource:
+) -> Union[RunResource, BadRunResource]:
     run_id = row.id
     protocol_id = row.protocol_id
     created_at = row.created_at
-
+    # Checking the fundamental data types here are not covered by the error handling
+    # because if they fire, the only thing we can do to address the issue is immediately
+    # delete the row while we still have a handle on it from sql - we won't have any
+    # other way to delete it. It's also unclear how it could happen without the table schema
+    # changing out from under us.
     assert isinstance(run_id, str), f"Run ID {run_id} is not a string"
     assert protocol_id is None or isinstance(
         protocol_id, str
     ), f"Protocol ID {protocol_id} is not a string or None"
-
-    return RunResource(
-        run_id=run_id,
-        created_at=created_at,
-        protocol_id=protocol_id,
-        actions=[
+    try:
+        actions = [
             RunAction(
                 id=action_row.id,
                 createdAt=action_row.created_at,
                 actionType=RunActionType(action_row.action_type),
             )
             for action_row in action_rows
-        ],
+        ]
+    except BaseException as be:
+        return BadRunResource(
```

Comment:
It probably makes sense to log a warning here, right?

```suggestion
        _log.warning("Error reading actions for run ID {run_id}:", exc_info=True)
        return BadRunResource(
```

Reply:
Sure!